### ISSUE TITLE
SQL Errors

### ISSUE THREAD

**Issue by natew** [motion/orbit#266]
Making this an umbrella ticket to get the database issues out of the way I've been seeing.

The few I've seen are basically:

- [ ] Race: between desktop/web conditions lock up the database, likely requires Cordova driver patch
- [ ] Desktop restarts: disconnects web, attempting to close it can cause trouble
- [ ] Syncers: occasionally bug when syncing lots of stuff and cause the database to lock out

## Race

This happens pretty often. Not sure the exact conditions to replicate, but if you have the app running with a few syncers you'll see it. May just have to get to it once it replicates. Looks like:

![image](https://user-images.githubusercontent.com/12100/41680161-75c82428-7485-11e8-9a26-e62f9e0ab469.png)

## Desktop restarts

First is, when Desktop restarts the Web loses connection permanently. I attempted to fix this but I get a separate error there.

So basically you see this in Web:

![image](https://user-images.githubusercontent.com/12100/41879687-8ff31dd6-788f-11e8-8649-319a99a88291.png)

The connection is stored at `Root.connection` but if you call `.close()` you oftentimes get an error that says "cannot close during transaction". Of course the one time it worked was in creating this ticket. I have the connection logic in `Root.connectModels()`, which presumably you could call to-reestablish connection.

## Syncers

Again, happens occasionally when syncing a lot. This likely pops up during development of syncers so no need to trace it down right away.
